central/scannerdefinitions/handler/handler.go

```diff
@@ -321,28 +345,35 @@ func (h *httpHandler) openMostRecentDefinitions(uuid string) (file *os.File, mod
 	u := h.getUpdater(uuid)
 	u.Start()
 
+	toClose := func(f *vulDefFile) {
+		if file != f && f != nil {
+			utils.IgnoreError(f.Close)
+		}
+	}
+
 	// Open both the "online" and "offline", and save their modification times.
-	onlineFile, onlineTime, err := u.file.Open()
+	var onlineFile *vulDefFile
+	onlineOsFile, onlineTime, err := u.file.Open()
 	if err != nil {
 		return
 	}
-	offlineFile, offlineTime, err := h.offlineFile.Open()
+	if onlineOsFile != nil {
+		onlineFile = &vulDefFile{File: onlineOsFile, modTime: onlineTime}
+	}
+
+	defer toClose(onlineFile)
+	offlineFile, err := h.openOfflineBlob(ctx)
 	if err != nil {
-		utils.IgnoreError(onlineFile.Close)
 		return
 	}
+	defer toClose(offlineFile)
 
 	// Return the most recent file, notice that if both don't exist, nil is returned
```

Comment:
not sure the reason as for why `nil` is returned here is necessarily true anymore

Reply:
It is the same scenario as it was before.